.github/workflows/presubmit.yml

```diff
@@ -0,0 +1,49 @@
+name: Presubmit
+
+permissions:
+  contents: read
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build all specs
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Install required packages
+        run: |
+          sudo apt-get install -y libpango1.0-dev libwebp-dev ghostscript fonts-lyx jing libavalon-framework-java libbatik-java python3-pyparsing
+          sudo gem install asciidoctor -v 2.0.16
+          sudo gem install coderay -v 1.1.1
+          sudo gem install rouge -v 3.19.0
+          sudo gem install ttfunk -v 1.7.0
+          sudo gem install hexapdf -v 0.27.0
+          sudo gem install asciidoctor-pdf -v 2.3.4
+          sudo gem install asciidoctor-mathematical -v 0.3.5
+          sudo pip install pyparsing
+
+      - name: List git tag
+        run: |
+          git describe --tags --dirty
+
+      - name: Generate core specs (HTML and PDF)
+        run: |
+          python3 makeSpec -clean -spec core OUTDIR=out.core -j 5 api c env ext cxx4opencl
+          
+      - name: Generate core + extension specs (HTML)
+        run: |
+          python3 makeSpec -clean -spec khr OUTDIR=out.khr -j 12 html
```

Comment:
Why aren't you generating the PDFs as well?

Reply:
I was trying to more-or-less faithfully follow what the Travis script was doing previously.  It generated both PDF and HTML versions of the "core" specs, but only HTML versions of the "khr" specs.

Here are the relevant lines from the Travis script:

```
  - python3 makeSpec -clean -spec core OUTDIR=out.core -j 5 api c env ext cxx4opencl
  - python3 makeSpec -clean -spec khr OUTDIR=out.khr -j 12 html manhtmlpages
```

@oddhack any specific reason for this?

Reply:
Generating the PDF is in there just to verify the PDF pipeline is working. There aren't structural elements in the extension spec that would challenge it any more than the core spec, and PDF generation is by far the slowest operation, so I only do it once.

The -j 12 is because there are a lot of manhtmlpages to generate, each quite fast. Very unlikely we would actually get 12 cores of CI though, I think the GH runners only have two.

If there's some reason people need the PDF artifacts then they can be built, of course. But for Vulkan purposes, CI is just checking that the pipeline works - our publication artifacts are built outside of CI.

Reply:
Is this running as a pre-commit filter, as the YML filename suggests? If so, that's another reason to avoid PDF generation as much as possible since that lengthens the loop.

We've done OK in Vulkan just making spec generation part of the regular post-commit CI step and fixing things after the fact, FWIW.

Reply:
> Is this running as a pre-commit filter, as the YML filename suggests?

We currently run it on pushes and pull requests, so it's both a pre-commit filter and a post-commit validation.  I suppose we could separate that if needed, although the current job executes pretty quickly.  Looking at the latest run, it completed in ~3 minutes:  One minute of that was installing packages, and another minute was building the online reference pages.  Spec-wise, HTML generation took ~10 seconds, and HTML + PDF generation took ~35 seconds.